### ISSUE TITLE
`dicti` accepts unions but new values are not properly parsed

### ISSUE THREAD

**Issue by DefiDebauchery** [coldfix/pydicti#8]
In Python 3.9 ([PEP-584](https://peps.python.org/pep-0584/)), the `|` union and `|=` union assignment operators were added to dicts, so that values can effectively be added and updated simultaneously. However, while pydicti classes don't error when using these and existing values are updated as expected, new keys do not get processed into `__case` , preventing dicti access methods to fetch the value.

```py
from pydicti import dicti

di = dicti({'AsdF': 1, 'jkl': 2})
di |= {'AsdF': 10, 'XYZ': 3} # also tried as dicti({'AsdF': 10, 'XYZ': 3})

# --- 

>>> di
<<< dicti({'AsdF': 10, 'jkl': 2, 'XYZ': 3})

>>> di.get('asdf')
<<< 10

>>> di.get('xyz')
<<<

>>> di['xyz']
KeyError: 'xyz'

>>> di['XYZ']
KeyError: 'xyz'
```

Unfortunately I'm not well-versed enough in the low-level aspects of dicts to have an idea on how to _properly_ add support for this. 

Edit: I did notice that the `__ior__` dunder method is called when using `|=`; simply proxying that to `self.update()` (and adding `return self`, otherwise the variable strangely becomes None) seemed to work as I would expect.

**Comment by coldfix** [comment]
Hi, thanks for reporting! It's probably because dicti inherits from dict but didn't overwrite the `|` protocol so far, meaning the builtin from `dict` will be called, which apparently doesn't invoke `self.update` as a method lookup.

You're right, the correct way to fix this is probably to simply overwrite `__ior__` and `__or__` deferring to `self.update()`.

> and adding return self, otherwise the variable strangely becomes None) seemed to work as I would expect.

That's one of the peculiarities of the inplace update operator protocol in python. Basically `a |= b` is equivalent to `a = a.__ior__(b)` I think.

**Comment by coldfix** [comment]
Fixed in v1.2.1.

**Comment by DefiDebauchery** [comment]
Amazing work, thank you very much!
